modules/gapi/src/executor/last_value.hpp

```diff
@@ -25,17 +25,29 @@ namespace own {
 // if there was no value written since the last read).
 //
 // Again, the implementation is highly inefficient right now.
+
+template <typename T>
+using DropStrategy = std::function<bool(const T& t)>;
+template <typename T>
+using OptDropStrategy = cv::optional<DropStrategy<T>>;
+
 template<class T>
 class last_written_value {
     cv::util::optional<T> m_data;
 
     std::mutex m_mutex;
     std::condition_variable m_cond_empty;
+    OptDropStrategy<T> m_drop_strategy;
 
     void unsafe_pop(T &t);
 
 public:
-    last_written_value() {}
+    last_written_value() {};
+
+    last_written_value(DropStrategy<T>&& drop_strategy)
```

Comment:
So here... Shouldn't this be part of class template itself?

Reply:
Question is still relevant. Why don't make the drop strategy part of the class' template itself?

Reply:
Done